presto-main/src/main/java/com/facebook/presto/operator/scalar/MathFunctions.java

```diff
@@ -438,4 +439,37 @@ else if (operand >= upper) {
 
         return result;
     }
+
+    @Description("The bucket number of a value given an array of bins")
+    @ScalarFunction("width_bucket")
+    @SqlType(StandardTypes.BIGINT)
+    public static long widthBucket(@SqlType(StandardTypes.DOUBLE) double operand, @SqlType("array(double)") Block bins)
+    {
+        int nbins = bins.getPositionCount();
+
+        checkCondition(nbins > 0, INVALID_FUNCTION_ARGUMENT, "Bins cannot be an empty array");
+        checkCondition(!Double.isNaN(operand), INVALID_FUNCTION_ARGUMENT, "Operand cannot be NaN");
+
+        int lower = 0;
+        int upper = nbins;
+
+        int index;
+        double bin;
+
+        while (lower < upper) {
```

Comment:
We may want to add checks to detect if the bins are not properly sorted (e.g., if bins[lower] is ever > bins[upper])